### DISCUSSION TITLE
Delete next stage

### DISCUSSION THREAD

**Discussion by Esmolante** [organicmaps/organicmaps#3995] [Ideas]
I suggest to split "Stop" button in navigation screen in three:
1 - delete route
2 - delete next stage (very useful when we pass close enough to the stage but the programme does not recognise it, and insists that we pass again).
3 - exit to edit route.
In general, OM lacks a convenient way to edit the route once it has been created. For example, when we want one stage to go before another.

On the other hand, the settings button could be removed from the panel, because it is not common to go to it while driving.

**Comment by biodranik** [comment]
1. What should happen if a route is deleted? How is it different from (3)?
2. Let's focus on fixing it instead. Can you please collect specific cases with specific routes, where the intermediate stop is not deleted (but it should)? You can create an issue for that, with all the necessary details and links to OpenStreetMap.org. It will help us to fix it.
3. OM allows editing routes on iOS, but not when in navigation mode. How do you see it in a convenient way during navigation? How other apps are doing it?
4. Now "Avoid..." options can be reached only via that settings button. So it is required.

**Comment by Esmolante** [comment]
1. Delete route: You exit to the map screen, without any planned route, clean. You use it when the navigation is already finished.
Edit route: you exit to a route editing screen, where you can reorder the routes, delete them or add new ones. With a button to access a view of the route on the map, where you can add and delete stages visually.
2. It is not necessarily a malfunction. In fact, OM works very well at closing a stage even if we haven't just passed over it. An example happened to me this morning: I dropped my wife off at the doctor's while I was still driving to another destination. I left her a couple of streets before to continue to my destination, but it was a very cumbersome process to tell OM that I no longer wanted to go to that intermediate stage. In fact, I had to delete the route and put it back in. A problem, when you're driving. And a bitch, if the route is complex, and there are several more stages to go (a professional, delivery driver).
3. Honestly, is not a task that can be performed while driving. It is necessary to turn off the road and retrace the route. So it should not be part of the navigation screen. But in navigation there may be a button to direct you to the edit screen, which would be as I described above. I don't know how you have solved it on iOS, but it is important for creating complex routes. 
4. I see. It might be too complicated to navigate between the options while driving. A button could lead directly to the permitted/prohibited road type dialogue.
